### ISSUE TITLE
sqitch add not processing variables in templates

### ISSUE THREAD

**Issue by psabo** [sqitchers/sqitch#684]
Running sqitch 1.3.1 with perl v5.16.3 and Template::Tiny 1.14 on an ec2 running amazon linux 2. 

Very simple template (only including deploy for simplicity)

```
-- Deploy media_reporting:[% ENGAGEMENT_NICE_ID %]_create_schema snowflake

USE WAREHOUSE &warehouse;

-- create and grant usage for [% ENGAGEMENT_NICE_ID %] engagement schema
CREATE SCHEMA IF NOT EXISTS &{region}_&{environ}_REPORTING.[% ENGAGEMENT_NICE_ID %];
```

Exact command being ran (with output):

```
sqitch add 2022-17-12_BLAH5_create_schema --template-directory templates --template create_schema --set ENGAGEMENT_NICE_ID=BLAH5 -n "BLAH5: sqitch add from template: create_schema"
Created deploy/2022-17-12_BLAH5_create_schema.sql
Created revert/2022-17-12_BLAH5_create_schema.sql
Created verify/2022-17-12_BLAH5_create_schema.sql
Added "2022-17-12_BLAH5_create_schema" to sqitch.plan
```

The right things appear to happen, all of the files are generated.. except the variable set on cli does not get populated. A diff on the "processed" file vs the template shows no difference. I'm not really sure how to troubleshoot this. The same command works fine locally on a Mac with the same repo.

Commands to install sqitch were:

```
sudo yum install -y @development zlib-devel bzip2 bzip2-devel readline-devel \
    sqlite sqlite-devel openssl-devel xz xz-devel libffi-devel findutils \
    openblas-devel wget awscli nano python3 python3-devel tmux gcc git \
    perl-devel perl-CPAN perl-DBD-ODBC libiodbc unixODBC-devel
```

```
curl -L https://cpanmin.us | perl - --sudo App::cpanminus
cpanm --sudo --quiet --notest --sudo App::Sqitch DBD::ODBC
sudo yum install -y libiodbc
```

Any advice/guidance would be greatly appreciated. 

edit: formatting.

**Comment by theory** [comment]
Replicated with AmazonLinux 2 in Docker. Investigating;

**Comment by theory** [comment]
I was able to get it to work by using lowercase variable names instead of uppercase. This might be a limitation of Template::Tiny. The uppercase variable names worked when I installed Template::Toolkit, which Sqitch will silently use if it is installed. I'll bet it's installed on your Mac. 

Ah yes, here is a [Template::Tiny bug report](https://rt.cpan.org/Public/Bug/Display.html?id=102767) for this issue (dating from 2015!). I therefore recommend either using lowercase variables (which nicely distinguishes variables from the Template language keywords IMO) or just install Template::Toolkit.
